### DISCUSSION TITLE
what flags or parameters would help with stuttering? HLS to UDP

### DISCUSSION THREAD

**Discussion by eriktovarj** [bluenviron/mediamtx#2315] [Questions]
Thanks for the support I have received, mainly from @aler9. 
Which flags can be used to help with stuttering?
The spurce HLS video when viewed in VLC or any browser doesn't have any problems or stutters, only when ffmpeg does the conversion from HLS to UDP?
TIA

**Comment by aler9** [comment]
Hello, what do you mean by stuttering? is there any error inside VLC logs or FFmpeg logs?

anyway, keep in mind that UDP is extremely fragile when routed through the internet, and packets can get lost very easily. There's not much you can do, except maybe re-encoding the video with a lower bitrate. If you can change protocol, there's a wrapper around UDP that is called SRT and improves UDP with security and reliability, by retransmitting lost packets. You can use it by following instructions in the README.

**Reply by eriktovarj** [reply]
by stuttering i mean interrupted audio and video. I think that has beed resolved using the - b (buffer), after reading the ffmepg documentation.

Now the problem resides in the fact that during the transmission ADs or commercials are inserted, just when the commercial is shown the transmission pauses and I must restart the command to have the video again. What parameter can I use to accept that redirection? VLC does not present these pauses when the movie or program is paused and the ad is displayed

**Reply by eriktovarj** [reply]
BTW, UDP will not hit the internet, i will be routed/used locally

**Reply by aler9** [reply]
ADs in HLS are inserted through features at the protocol level, but unfortunately these features are not supported by gohlslib (HLS library behind MediaMTX), mainly due to the lack of publicly available streams that can be used for development.

**Reply by eriktovarj** [reply]
any way to restart the command automatically if a problem is found in the transmission?

what is runOnReadyRestart used for?

**Reply by eriktovarj** [reply]
ADs in HLS are inserted through features at the protocol level, but unfortunately these features are not supported by gohlslib (HLS library behind MediaMTX). Does the samething happend to ffmpeg?

im only using ffmpeg and not mediaMTX as you had suggested

**Reply by aler9** [reply]
> im only using ffmpeg and not mediaMTX as you had suggested

Hence the ADS features are not supported by FFmpeg too. I'm referring to the "interstitials" feature of HLS, documented here:

https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#appendix-D

> any way to restart the command automatically if a problem is found in the transmission?

yes, if you're using Linux you can wrap the command inside while:

```
while true; do ffmpeg -i ....; done
```

Regarding Windows, there is surely a BAT-based equivalent, just search it online.

[a comment by an automated account is not shown]
